src/busctl/busctl.c

```diff
@@ -1201,44 +1199,19 @@ static int message_pcap(sd_bus_message *m, FILE *f) {
 }
 
 static int message_json(sd_bus_message *m, FILE *f) {
-        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL, *w = NULL;
-        char e[2];
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
         int r;
-        usec_t ts;
 
-        r = json_transform_message(m, &v);
+        assert(m);
+
+        r = sd_bus_message_dump_json(m, SD_BUS_MESSAGE_DUMP_WITH_HEADER, &v);
         if (r < 0)
-                return r;
+                return log_error_errno(r, "Failed to build JSON object from DBus message: %m");
+
+        r = sd_json_variant_dump(v, arg_json_format_flags, f, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to show JSON object: %m");
```

Comment:
Not your fault, but message_dump and message_pcap don't log for error. Better to make things consistent if you touch this?